src/sql-parser/src/parser.rs

```diff
@@ -1305,6 +1307,32 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_at_most_one_keyword(
+        &mut self,
+        a: Keyword,
+        b: Keyword,
+        location: &str,
+    ) -> Result<ParseAtMostOneResult, ParserError> {
```

Comment:
This is a nice addition! Rather than adding `ParseAtMostOneResult`, I think this could look a little more like `parse_one_of_keywords`. Something like

```rust
    fn parse_at_most_one_keyword(
        &mut self,
        kws: &[Keyword],
        location: &str,
    ) -> Result<Option<Keyword>, ParserError> {
        let first = self.parse_one_of_keywords(kws);
        let second = self.parse_one_of_keywords(kws);

        match (first, second) {
        ...
```

This has the advantage of returning the actual keyword rather than an enum, and generalizes to > 2 inputs.

Reply:
chatted offline but I'll summarize:

In general I like avoiding writing functions that make it possible to return non-sensical values.  e.g., here, `Some(keyword_not_in_the_kws_list)` doesn't make sense to return but is allowed by the type signature.  Given all three current use cases are `match!(self.parse_at_most_one_keyword, Some(keyword))`, I think it'll be okay to match other functions in this file.